File: app/src/main/java/com/example/myfakequotesapp/Presentation/View/Adapters/QuoteListAdapter.java
package com.example.myfakequotesapp.Presentation.View.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myfakequotesapp.Domain.Model.Quote;
import com.example.myfakequotesapp.R;

import java.util.List;

public class QuoteListAdapter extends RecyclerView.Adapter<QuoteListAdapter.QuoteViewHolder>{
    private Context context;
    List<Quote> QuoteList;

    public QuoteListAdapter(Context context){
        this.context = context;
    }

    public void setQuoteList(List<Quote> quoteList){
        this.QuoteList = quoteList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public QuoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).
                inflate(R.layout.quote_list_element, parent, false);

        return new QuoteViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull QuoteViewHolder holder, int position) {
        holder.quoteText.setText(this.QuoteList.get(position).quoteText);
        holder.nameAuthor.setText(this.QuoteList.get(position).nameAuthor);
        holder.lastnameAuthor.setText(this.QuoteList.get(position).lastnameAuthor);

    }

    @Override
    public int getItemCount() {
        if (QuoteList == null) {
            return 0;
        }
        else {
            return QuoteList.size();
        }
    }
    public class QuoteViewHolder extends RecyclerView.ViewHolder{

        TextView quoteText;
        TextView nameAuthor;
        TextView lastnameAuthor;
        public QuoteViewHolder(View view){
            super(view);
            quoteText = view.findViewById(R.id.quoteTextElementView);
            nameAuthor = view.findViewById(R.id.nameAuthorTextElementView);
            lastnameAuthor = view.findViewById(R.id.lastnameAuthorTextElementView);
        }
    }
}

File: app/src/main/java/com/example/myfakequotesapp/View/Adapters/QuoteListAdapter.kt
package com.example.myfakequotesapp.View.Adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.myfakequotesapp.Model.Quote
import com.example.myfakequotesapp.R
import com.example.myfakequotesapp.View.Adapters.QuoteListAdapter.QuoteViewHolder

class QuoteListAdapter(private val context: Context) : RecyclerView.Adapter<QuoteViewHolder>() {
    //var quoteList: List<Quote?>? = null
    internal var quoteList: List<Quote?>? = null
    private var listener: OnItemClickListener? = null
    fun getQuoteList(): List<Quote?>? {
        return quoteList
    }

    fun setQuoteList(quoteList: List<Quote?>?) {
        this.quoteList = quoteList as MutableList<Quote?>?
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): QuoteViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.quote_list_element, parent, false)
        return QuoteViewHolder(view)
    }

    override fun onBindViewHolder(holder: QuoteViewHolder, position: Int) {
        holder.quoteText.text = quoteList!![position]!!.quoteText
        holder.nameAuthor.text = quoteList!![position]!!.nameAuthor
        holder.lastnameAuthor.text = quoteList!![position]!!.lastnameAuthor
        holder.title.text = quoteList!![position]!!.title
    }

    override fun getItemCount(): Int {
        return if (quoteList == null) {
            0
        } else {
            quoteList!!.size
        }
    }

    inner class QuoteViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var quoteText: TextView
        var nameAuthor: TextView
        var lastnameAuthor: TextView
        var title: TextView
        var onItemClickListener: OnItemClickListener? = null

        init {
            quoteText = view.findViewById(R.id.quoteTextElementView)
            nameAuthor = view.findViewById(R.id.nameAuthorTextElementView)
            lastnameAuthor = view.findViewById(R.id.lastnameAuthorTextElementView)
            title = view.findViewById(R.id.textViewTitleElementView)
            view.setOnClickListener {
                val position = adapterPosition
                if (listener != null && position != RecyclerView.NO_POSITION) {
                    listener!!.onItemClick(quoteList!![position])
                }
            }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(quote: Quote?)
    }

    /*fun setOnItemClickListener(listener: (Any) -> Unit) {
        this.listener = listener
    }*/
    open fun setOnItemClickListener(listener: (Quote) -> Unit) {
        this.listener = listener as OnItemClickListener?
    }
}